### ISSUE TITLE
Form Builder: Virtual form fields that do not directly map to a database column

### ISSUE THREAD

**Issue by dtdesign** [WoltLab/WCF#2988]
Scenario: Having a single column that is mapped to two virtual form fields. For example, consider the column `fullname` that is presented as two separate input fields: `firstname` and `lastname`.

We need some sort of callbacks (?) to reliably accomplish these tasks:
1. Provide values for those virtual form fields.
2. Modify the values that are provided for the DBOAction, e.g. joining the columns in the example.
3. Set the proper values when editing an object, currently the old values are displayed, because the system is confused by the lack of a direct mapping.

**Comment by Gravatronics** [comment]
> 2. Modify the values that are provided for the DBOAction, e.g. joining the columns in the example.

That is already possible with a custom`IFormFieldDataProcessor` implementation.

> 3. Set the proper values when editing an object, currently the old values are displayed, because the system is confused by the lack of a direct mapping.

I am wondering if this could/should be added to `IFormFieldDataProcessor` so that such processors not only handle the conversion from form data to DBO action parameters but also from DBO object to form field value.

> 1. Provide values for those virtual form fields.

I am unsure if that should really be supported by forms themselves. I would rephrase the original statement

> Having a single column that is mapped to two virtual form fields.

to focus on the perspective of the form:

> Having two form fields that are mapped to a single database column/DBO property.

**Comment by mutec** [comment]
How can I actually unset the value/field of a field?
Example: I'm getting the username by `UserFormField` but don't want to store the username itself, but the user's id. I can set the new field by adding it to `AbstractFormBuilderForm::$additionalFields` but I can't remove the username-field without adding a `FormDataProcessor` and searching for a field with this specific id. Did I miss an easier way?
